include/seqan3/core/bit_manipulation.hpp

```diff
@@ -17,11 +17,20 @@
 
 #include <utility>
 
-#include <seqan3/core/platform.hpp>
+#include <seqan3/core/detail/int_types.hpp>
+#include <seqan3/std/concepts>
 
 namespace seqan3::detail
 {
 
+/*!\brief How many bits has a type?
+ * \ingroup core
+ *
+ * \tparam type_t The type to determine the number of bits..
+ */
+template <typename type_t>
+constexpr auto max_bits = min_viable_uint_v<8u * sizeof(type_t)>;
```

Comment:
Maybe we can call it something like `bit_size`?

`max_bits` kinda implies dynamic size to me.

Reply:
I changed the name to `sizeof_bits` I hope this makes it more clear